include/sqlgen/select_from.hpp

```diff
@@ -268,7 +268,7 @@ struct SelectFrom {
     static_assert(std::is_same_v<LimitType, Nothing>,
                   "You cannot call limit twice.");
       return SelectFrom<TableOrQueryType, AliasType, FieldsType, JoinsType,
-                      WhereType, GroupByType, OrderByType, Limit, Offset, ToType>{
+                      WhereType, GroupByType, OrderByType, LimitType, OffsetType, ToType>{
         .fields_ = _s.fields_,
```

Comment:
If I understand Qwen's comments correctly, it seems this was not entirely correct for Limit as well:

> In both operator| overloads for Limit and Offset, you must preserve the original OffsetType or LimitType when they are not being replaced.
> [...]
>  # Why This Works
> SelectFrom is parameterized by types, not values.
> When you haven’t called .offset(...), the OffsetType is rfl::Nothing, and .offset_ is a rfl::Nothing instance.
> When you do call .offset(n), it returns a new SelectFrom with OffsetType = sqlgen::Offset (or similar), and .offset_ holds the value.
> The operator| overloads must carry forward the existing types for unchanged components.